### PR TITLE
 Modules structure. Templates. Modal. Refactoring

### PR DESCRIPTION
**Pull request by f3arnil**
1. Final modules structure
2. Templates ( now use in Header \ footer \ simple search \ advanced search \ error in promises)
3. Modal window module with example in search simple and search advanced
4. Refactoring

### CODE REVIEW

File: angular-spa-ui/js/common/modal/index.html [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<script type="text/ng-template" id="myModalContent.html">
+        <div class="modal-header">
```
Comment (tarhunakau) [inline angular-spa-ui/js/common/modal/index.html]: Setup codestyle in your IDE... 1 tab symbol should be replaced with 4 spaces. I see 8.

File: angular-spa-ui/js/search-app/bootstrap.js [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 module.exports = function(app){
```
Comment (tarhunakau) [inline angular-spa-ui/js/search-app/bootstrap.js]: Setup codestyle in your IDE. I do not see the space character after "(app)" and before "{".

File: angular-spa-ui/js/search-app/cart-module/index.js [hunk 3]
```diff
@@ -1,14 +1,16 @@
...
+    
+    cart.config(configCb);
+    
     function configCb($stateProvider) {
         $stateProvider
             .state('cart', {
                 url: '/cart',
-                template: 'Hello cart'
+                template: 'Hello world from cart module',
             })
```
Comment (tarhunakau) [inline angular-spa-ui/js/search-app/cart-module/index.js]: ";" is missing.

File: angular-spa-ui/js/search-app/error.html [hunk 4]
```diff
@@ -1,6 +1,8 @@
 <script type="text/ng-template" id="/error.html">
+
```
Comment (tarhunakau) [inline angular-spa-ui/js/search-app/error.html]: Why did you add 2 empty lines (2nd and 7th lines) in the template?

File: angular-spa-ui/js/search-app/search-app-ctrl.js [hunk 5]
```diff
@@ -2,20 +2,20 @@
...
+    app.controller('mainCtrl', function( $scope, promises, getTemplate, appConfig) {
 
         var config = appConfig.config;
-        var path = config.path;
+        var paths = config.paths;
         var methods = config.methods;
         var userData = config.userData;
         
-        path.getUserPath = path.getUserPath + GLOBAL_USER_ID;
+        paths.userPath = paths.userPath + GLOBAL_USER_ID;
```
Comment (tarhunakau) [inline angular-spa-ui/js/search-app/search-app-ctrl.js]: variable = variable + data
variable += data

What is better?

File: angular-spa-ui/js/common/modal/index.html (hunk 1 above)
Comment (f3arnil) [inline angular-spa-ui/js/common/modal/index.html]: Its not IDE, this code is from web. I'll fix it

File: angular-spa-ui/js/search-app/search-app-ctrl.js [hunk 6]
```diff
@@ -27,41 +27,48 @@ module.exports = function(app) {
...
+                        return getTemplate.getByTrustedUrl('/footer.html', element, $scope);
+                    })
+                    .catch(
+                        function(err) {
+                        console.log(err);
+                    });
                 })
             .catch(
-                function(error){
+                function(error) {
```
Comment (tarhunakau) [inline angular-spa-ui/js/search-app/search-app-ctrl.js]: Why are you ignoring codestyle?
function_____a(variable)_____{
...
var_____b_____=_____function_____(...)_____{

File: angular-spa-ui/js/search-app/error.html (hunk 4 above)
Comment (f3arnil) [inline angular-spa-ui/js/search-app/error.html]: To make template more readable. If critical I'll fix it

File: angular-spa-ui/js/search-app/search-app-ctrl.js [hunk 7]
```diff
@@ -27,41 +27,48 @@ module.exports = function(app) {
...
                     $scope.inited = false;
                     $scope.err = { code : error.status, data : error.data};
-                    var element = angular.element( document.querySelector( '#main' ) );
+                    var element = angular.element( document.querySelector( '#app' ) );
```
Comment (tarhunakau) [inline angular-spa-ui/js/search-app/search-app-ctrl.js]: Codestyle discrepancy

File: angular-spa-ui/js/search-app/search-module/index.jade [hunk 8]
```diff
@@ -0,0 +1 @@
+include search.html
```
Comment (tarhunakau) [inline angular-spa-ui/js/search-app/search-module/index.jade]: Did you review the pull request? Or this red arrow and sign are visible only for me?

File: angular-spa-ui/package.json [hunk 9]
```diff
@@ -1,27 +1,28 @@
...
-        "url-loader": "*",
-        "file-loader": "*",
-        "style-loader": "*",
-        "webpack": "^1.12.3"
-    },
-    "scripts": {
-        "dev": "set NODE_ENV=development&& webpack",
-        "prod": "webpack"
-    }
+  "name": "angular-spa-ui",
```
Comment (tarhunakau) [inline angular-spa-ui/package.json:2]: Codestyle discrepancy 99% of code, fake commit.

**Comment by tarhunakau** [comment]
Setup codestyle on your IDE

File: angular-spa-ui/js/search-app/search-app-ctrl.js (hunk 5 above)
Comment (f3arnil) [inline angular-spa-ui/js/search-app/search-app-ctrl.js]: Line 17 -18 -> I fix it, here a miss it
Will fix

File: angular-spa-ui/js/search-app/search-app-ctrl.js (hunk 6 above)
Comment (f3arnil) [inline angular-spa-ui/js/search-app/search-app-ctrl.js]: May be you want to say that "function(error) {" should be "function (error) {" ?

File: angular-spa-ui/js/search-app/search-module/search.html [hunk 10]
```diff
@@ -0,0 +1,16 @@
+<script type="text/ng-template" id="/search.html">
+
```
Comment (tarhunakau) [inline angular-spa-ui/js/search-app/search-module/search.html:2]: I missed it, remove empty lines from this file too

File: angular-spa-ui/js/search-app/cart-module/index.js [hunk 11]
```diff
@@ -1,15 +1,17 @@
 "use strict";
 
-module.exports = function(app) {
-
-    app.config(configCb);
+module.exports = function (angular) {
 
+    var cart = angular.module('searchApp.Cart',[]);
```
Comment (tarhunakau) [inline angular-spa-ui/js/search-app/cart-module/index.js]: No space between ",[" symbols

File: angular-spa-ui/js/search-app/search-module/advanced-search/index.js [hunk 12]
```diff
@@ -1,15 +1,24 @@
 "use strict";
 
-module.exports = function(app) {
-
-    app.config(configCb);
+module.exports = function (angular) {
 
+    var advancedSearch = angular.module('searchApp.Search.advanced',[]);
```
Comment (tarhunakau) [inline angular-spa-ui/js/search-app/search-module/advanced-search/index.js]: No space between ",[" symbols

**Comment by tarhunakau** [comment]
If we ignoring code styles at the moment, looks good from my side and could be merged
